### DISCUSSION TITLE
20230918 update: JSON error generated when pressing Addons button

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1430] [Fixed Issues]
Running rAudio 20230620

JSON error is immediately generated when pressing the Addons button:

[20230918 Json Error.txt](https://github.com/rern/rAudio/files/12660903/20230918.Json.Error.txt)

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by Raitsa** [comment]
https://i.imgur.com/nfD0fvu.png

**Comment by kw-sudo** [comment]
@Sagnasty 

I had the same issue, that one helped: https://github.com/rern/rAudio/discussions/1421#discussioncomment-7037026

RasPi 3B+
DS716+II

Oliver

**Comment by Maikeleg** [comment]
> Running rAudio 20230620
> 
> JSON error is immediately generated when pressing the Addons button:
> 
> [20230918 Json Error.txt](https://github.com/rern/rAudio/files/12660903/20230918.Json.Error.txt)
> 
> Jim
> 
> 3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro 1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro Synology DS718+

Hi Jim,
I have exactly the same error message as you. For me the suggestion from @master-kw Oliver didn't work
Did you found a sollution?

Maikel

**Comment by rern** [comment]
Please try: https://github.com/rern/rAudio/discussions/1421#discussioncomment-7037026

**Comment by Maikeleg** [comment]
> Please try: [#1421 (comment)](https://github.com/rern/rAudio/discussions/1421#discussioncomment-7037026)

Hi rern,
I did and i think it is also changed;
in line 256 in /srv/http/bash/common.sh
"pushstream() {
        local channel data ip json path sharedip updatedone webradiocopy"

However the error is the same, I'm on 20230630
executed with sudo

**Comment by marcelgsteiger** [comment]
I had exactly the same error here - running 20230630 on rpi4/4GB w/hifiberry dac+ pro and accessing through a browser in external net.

With the local browser on RPI on the same box, the error message is slightly different:

`JSON Error: no number after minus sign in JSON at position 2424`

but points to same position (the letter e) in the line containing the "status" / "hide" stanza.

When I ssh as root and cd /srv/http, then execute 

`/srv/http/bash/settings/addons-data.sh >temp.tmp`
and feed the result to jsonlint.com, I get the error on line 3 column 25 confirmed as follows:
`{ "push" : { "bash" : "" , "url" : "" , "hide" : 1 } , "r1" : { "title" : "rAudio" , "version" : "20231001" , (...)`
`	, "status" : {`
`		  "hide"        : [[ -e /boot/kernel.img ]] && echo 1]`
`		, "installed"   : ["r1"]`
`		, "notverified" : [""]`
`		, "update"      : ["r1"]`
`		, "online"      : true`
`	}`
`}`

`Error: Parse error on line 3:`
`... "hide"        : [[ -e /boot/kernel.img
`
`-----------------------^`
`Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', ']', got 'undefined'`

To me it seemed that something goes wrong while setting up the $hide variable for the addons.

As a quick and dirty workaround, I disabled the "hiding" of addons by changing addons-data.sh as follows:

`44c44`
`< 		  "hide"        : '$hide'`
`---`
`> 		  "hide"        : '[""]'`

This made the JSON error disappear - I could then upgrade to version 20231001 without any problems. The update replaced the file addons-data.sh again - undoing the dirty fix. The problem does not appear anymore.

HTH anyone.

Regards
Marcel

**Reply by Maikeleg** [reply]
Thanks a lot Marcel,
That did the trick, and the update also went flawless after that
Cheers, Maikel

**Comment by Sagnasty** [comment]
Thanks for the suggestions, but I was not so lucky ;-)

I rebuilt all my systems directly from image 20231022 with no further issues.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+
